admin/javascript/src/tests/silverstripe-backend-test.js

```diff
@@ -163,5 +163,29 @@ describe('SilverStripeBackend', () => {
         expect(result).toEqual({ status: 'ok', message: 'happy' });
       });
     });
+
+    it('should replace url template parameters', () => {
+      const mock = getBackendMock({
+        text: () => Promise.resolve('{"status":"ok"'),
+        headers: new Headers({
+          'Content-Type': 'application/json',
+        }),
+      });
+      const endpoint = mock.createEndpointFetcher({
+        url: 'http://example.com/:type/:id/?foo=bar',
+        method: 'post',
+        urlTemplateParams: {
+          ':id': 'id',
+          ':type': 'type',
+        },
+      });
+      const promise = endpoint({
+        id: 1,
+        type: 'mytype',
+        other: true
+      });
+      expect(mock.post.mock.calls[0][0]).toEqual('http://example.com/mytype/1/?foo=bar');
+      expect(mock.post.mock.calls[0][1]).toEqual('id=1&type=mytype&other=true');
```

Comment:
I find my `expect()` calls more helpful. `expect().toBeCalledWith()` simply says it hasn't been called with that combination of arguments, which isn't entirely helpful - it might mean the method doesn't get called at all, or that you're one character off in an argument somewhere.